charts/baseChart.js

```diff
@@ -40,17 +58,34 @@ module.exports = class Chart extends React.Component {
     }
 
     /**
+     * @inheritdoc
+     */
+    componentDidMount() {
+        const config = this.props.config;
+
+        this._retrieveResultsHelper(config.resultsUrl, config.resultsFunc)
+            .then(jsonResult => this.setState({ data: jsonResult }))
+            .catch(err => this.setState({ error: err }));
```

Comment:
just a warning, and one that is for thought and not action at this point, this is mixing data handling and presentation logic. Normally the two should be separated, if convenient, to avoid having the smarts of the application "hidden" within what should be simpler presentation logic. Ideally presentation logic just takes in data and displays it in some manner.

Reply:
if there's a better way to trigger the data retrieval let me know. this was based on an example i'd found.